src/manifest.c

```diff
@@ -65,17 +138,25 @@ static gboolean parse_manifest(GKeyFile *key_file, RaucManifest **manifest, GErr
 
 			image->slotclass = g_strdup(groupsplit[1]);
 
-			value = g_key_file_get_string(key_file, groups[i], "sha256", NULL);
+			value = manifest_consume_string(key_file, groups[i], "sha256", NULL);
 			if (value) {
 				image->checksum.type = G_CHECKSUM_SHA256;
 				image->checksum.digest = value;
 			}
 			image->checksum.size = g_key_file_get_uint64(key_file,
 					groups[i], "size", NULL);
+			g_key_file_remove_key(key_file, groups[i], "size", NULL);
 
-			image->filename = g_key_file_get_string(key_file, groups[i], "filename", NULL);
+			image->filename = manifest_consume_string(key_file, groups[i], "filename", NULL);
 
 			raucm->images = g_list_append(raucm->images, image);
+
```

Comment:
Shouldn't we check for remaining keys and remove the group at the end of each loop iteration? Currently we don't check/remove groups with RAUC_FILE_PREFIX.